### PR TITLE
deps: migrate to Poetry & pyproject.toml

### PR DESCRIPTION
**Pull request by mochaaP**
### Which kind of PR do you create?

- [x] This PR only contains minor fixes.
- [ ] This PR contains major feature update.
- [ ] This PR introduces a new function/API for Qiling Framework.

### Coding convention?

- [ ] The new code conforms to Qiling Framework naming convention.
- [ ] The imports are arranged properly.
- [ ] Essential comments are added.
- [ ] The reference of the new code is pointed out.

### Extra tests?

- [ ] No extra tests are needed for this PR.
- [ ] I have added enough tests for this PR.
- [x] Tests will be added after some discussion and review.

### Changelog?

- [ ] This PR doesn't need to update Changelog.
- [x] Changelog will be updated after some proper review.
- [ ] Changelog has been updated in my PR.

### Target branch?

- [x] The target branch is dev branch.

### One last thing

- [x] I have read the [contribution guide](https://docs.qiling.io/en/latest/contribution/)

-----

Migrate the project to [Poetry](https://python-poetry.org).

Supersedes #1111.

### CODE REVIEW

**Comment by mochaaP** [comment]
cc @xwings please review :)

**Review by xwings — APPROVED** [review]

**Comment by xwings** [comment]
code looks good to me. one more approve we are good to go

**Comment by mochaaP** [comment]
Any ideas on the eth_abi v4 thing? I literally have zero experience with Ethereum, updating to the new `encode` API resulted in a different bytecode being emitted.

**Comment by mochaaP** [comment]
@nullableVoidPtr Hi, do you have time to take a look at this, since you authored #1111?

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+py-ecc = { version = ">=1.4.7", optional = true }
+rlp = { version = "*", optional = true }
+trie = { version = "*", optional = true }
+numpy = { version = "*", optional = true }
+rich = { version = "*", optional = true }
+cmd2 = { version = "*", optional = true }
+eth-hash = { extras = ["pycryptodome"], version = "*", optional = true }
+
+unicornafl = { version = "^2.0.0", markers = "platform_system != 'Windows'", optional = true }
+fuzzercorn = { version = "^0.0.1", markers = "platform_system == 'Windows'", optional = true }
```
Comment (wtdcode) [inline pyproject.toml:40]: fuzzercorn should be installed on Linux only.

**Comment by wtdcode** [comment]
Looks good overall, but separating linting in a single commit would be fine. Anyway nice work!

This PR reminds me one thing, maybe we should really mention we only deal with `python>=3.8` somewhere.

**Comment by wtdcode** [comment]
Just one more thing, could you test if the release CI breaks or not?

**Comment by mochaaP** [comment]
> Just one more thing, could you test if the release CI breaks or not?

I tested it on my local machine with a dry run, no issue so far.

File: pyproject.toml (hunk 1 above)
Comment (mochaaP) [reply pyproject.toml]: Rebased and fixed.

**Review by xwings — APPROVED** [review]

**Comment by xwings** [comment]
CI seems to be good. @wtdcode you might need to approve the code

**Review by wtdcode — APPROVED** [review]

**Pull request merged**